example/ipsec_api/odp_ipsec.c

```diff
@@ -640,11 +640,20 @@ pkt_disposition_e do_ipsec_in_classify(odp_packet_t *ppkt)
 		return PKT_POSTED;
 	} else {
 		int out = 1;
+		odp_ipsec_packet_result_t result;
 
 		rc = odp_ipsec_in(ppkt, 1, ppkt, &out, &in_param);
 		if (rc <= 0)
 			return PKT_DROP;
 
+		if (odp_ipsec_result(&result, *ppkt) < 0) {
```

Comment:
So even in sync mode, we need to call odp_ipsec_result() to check if the IPsec processing was successful or not. Seeing this, I was wondering if the odp_ipsec_result() API should be made as a mandatory call in sync mode as well. What is your thoughts on that?

I recall seeing some related discussion in https://github.com/OpenDataPlane/odp/pull/1095.

Reply:
I do not quite see the need for making odp_ipsec_result() mandatory in sync mode.

For async mode the rationale was that for a HW offloaded implementation some SW post-processing may be needed and making odp_ipsec_result() mandatory avoids having to check for the post-processing need in odp_schedule() (when vectors are used and there is no separate event-to-packet conversion).

In sync mode the implementation should be able to easily perform the needed postprocessing within the synchronous call.

Reply:
Ack.